src/components/Post/Tabs/PostDescription.tsx

```diff
@@ -25,20 +29,76 @@ const CommentsContainer = dynamic(() => import('../Comment/CommentsContainer'),
 });
 
 interface IPostDescriptionProps {
+	allowed_roles?: string[] | null;
 	className?: string;
 	canEdit: boolean | '' | undefined;
 	id: number | null | undefined;
 	isEditing: boolean;
 	isOnchainPost: boolean;
+	trackName?: string;
+	isOffchainPost?: boolean;
 	toggleEdit: () => void
 	TrackerButtonComp: JSX.Element
 	Sidebar: ({ className }: {className?: string | undefined;}) => JSX.Element
 }
 
 const PostDescription: FC<IPostDescriptionProps> = (props) => {
-	const { className, canEdit, id, isEditing, toggleEdit, Sidebar, TrackerButtonComp } = props;
+	const { className, canEdit, id, isEditing, toggleEdit, Sidebar, TrackerButtonComp , allowed_roles, trackName , isOffchainPost } = props;
 	const { postData: { content, postType, postIndex, title, post_reactions } } = usePostDataContext();
-
+	const { network } = useNetworkContext();
+	const router = useRouter();
+	//write a function which redirects to the proposalType page
+	const deletePostFromUrl = (proposalType: ProposalType, trackName?: string) => {
```

Comment:
This logic is already in `BackToListingView.tsx` please extract that logic into a util function and use it here and there.

`getListingPathFromProposalType(proposalType:ProposalType)`

Reply:
`listingViewPath`